testsuite/tests/typing-local/local.ml

```diff
@@ -1774,3 +1774,40 @@ Line 1, characters 10-11:
               ^
 Error: This value escapes its region
 |}]
+
+(* Example of backtracking after mode error *)
+let f g n =
+  let a = local_ [n+1] in
+  let () = g a in
+  ()
+let z : (int list -> unit) -> int -> unit = f
+[%%expect{|
+val f : (local_ int list -> unit) -> int -> unit = <fun>
+Line 5, characters 44-45:
+5 | let z : (int list -> unit) -> int -> unit = f
+                                                ^
+Error: This expression has type (local_ int list -> unit) -> int -> unit
+       but an expression was expected of type
+         (int list -> unit) -> int -> unit
+       Type local_ int list -> unit is not compatible with type
+         int list -> unit
+|}]
+
+module M = struct
+  let f g n =
+    let a = local_ [n+1] in
+    let () = g a in
+    ()
+  let z : (int list -> unit) -> int -> unit = f
+end
+[%%expect{|
+Line 6, characters 46-47:
+6 |   let z : (int list -> unit) -> int -> unit = f
+                                                  ^
+Error: This expression has type
+         (local_ int list -> local_ unit) -> int -> unit
+       but an expression was expected of type
+         (int list -> unit) -> int -> unit
+       Type local_ int list -> local_ unit is not compatible with type
+         int list -> unit
```

Comment:
I'm not happy with the `local_ unit` in the return value here, but it's not to do with the change in this patch.

I think `oprint.ml` is printing a `Otyp_arrow (_, Oam_local, _, Oam_unknown, _)` with a local arg and local return, which isn't ideal.

Reply:
I think that:
```ocaml
print_out_type_1 mode ppf ty
```
should be:
```ocaml
print_out_type_1 rm ppf ty
```
on line 314 of `oprint.ml`

Reply:
Thanks, that makes sense. Patch applied.